logical/logical.go

```diff
@@ -47,6 +64,17 @@ type Backend interface {
 	// to the backend. The backend can use this to clear any caches or reset
 	// internal state as needed.
 	InvalidateKey(key string)
+
+	// Configure is used to set up the backend based on the provided backend
+	// configuration. This is primarily used in the the context of backend
+	// plugins, where the RPC server has no context of the core's config.
+	// This is the equivalent of calling framework.Backend.Setup(conf), but
+	// from the RPC client to configure components such as the logger, storage,
+	// system view, etc.
+	Configure(*BackendConfig) error
```

Comment:
I don't really want framework.Backend to have two functions that do the same thing. Let's consolidate to a single function in the `logical.Backend` interface and have `framework.Backend` implement that function. The question then is Setup or Configure. The function signature for Configure here is likely correct (you don't need Setup returning a logical.Backend when it's part of that interface), although it feels strange to me to call it Configure since it's not really configuring the stuff inside the backend...a bit misleading. My preference would be to call it Setup.